diff --git a/src/common.h b/src/common.h
index eb801bb..4aaace5 100644
--- a/src/common.h
+++ b/src/common.h
@@ -907,4 +907,6 @@ void finish_deferred_unlinks (void);
 /* Module exit.c */
 extern void (*fatal_exit_hook) (void);
 
+GLOBAL int debian_longlink_hack;
+
 _GL_INLINE_HEADER_END
diff --git a/src/create.c b/src/create.c
index e14e13d..3a84c79 100644
--- a/src/create.c
+++ b/src/create.c
@@ -27,6 +27,8 @@
 #include "common.h"
 #include <hash.h>
 
+extern int debian_longlink_hack;
+
 /* Error number to use when an impostor is discovered.
    Pretend the impostor isn't there.  */
 enum { IMPOSTOR_ERRNO = ENOENT };
@@ -724,7 +726,7 @@ write_header_name (struct tar_stat_info *st)
       return write_short_name (st);
     }
   else if (NAME_FIELD_SIZE - (archive_format == OLDGNU_FORMAT)
-	   < strlen (st->file_name))
+	   < strlen (st->file_name) + debian_longlink_hack)
     return write_long_name (st);
   else
     return write_short_name (st);
@@ -1476,7 +1478,7 @@ dump_hard_link (struct tar_stat_info *st)
 	  block_ordinal = current_block_ordinal ();
 	  assign_string (&st->link_name, link_name);
 	  if (NAME_FIELD_SIZE - (archive_format == OLDGNU_FORMAT)
-	      < strlen (link_name))
+	      < strlen (link_name) + debian_longlink_hack)
 	    write_long_link (st);
 
 	  st->stat.st_size = 0;
diff --git a/src/tar.c b/src/tar.c
index 4f5017d..e9a5126 100644
--- a/src/tar.c
+++ b/src/tar.c
@@ -1246,6 +1246,16 @@ expand_pax_option (struct tar_args *targs, const char *arg)
   return res;
 }
 
+/* Debian specific environment variable used by pristine-tar to enable use of
+ * longlinks for filenames exactly 100 bytes long. */
+void debian_longlink_hack_init () {
+ char *s=getenv ("TAR_LONGLINK_100");
+ if (s && strcmp(s, "1") == 0)
+	 debian_longlink_hack=1;
+ else
+	 debian_longlink_hack=0;
+}
+
 
 static uintmax_t
 parse_owner_group (char *arg, uintmax_t field_max, char const **name_option)
@@ -2626,6 +2636,8 @@ main (int argc, char **argv)
   filename_terminator = '\n';
   set_quoting_style (0, DEFAULT_QUOTING_STYLE);
 
+  debian_longlink_hack_init ();
+
   /* Make sure we have first three descriptors available */
   stdopen ();
 
